codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanDecoder.java

```diff
@@ -213,4 +139,117 @@ private static void insert(Node root, int symbol, int code, byte length) {
+            this.initialCapacity = initialCapacity;
+        }
+
+        void reset() {
+            node = ROOT;
+            current = 0;
+            currentBits = 0;
+            symbolBits = 0;
+            bytes = new byte[initialCapacity];
+            index = 0;
+        }
+
+        /*
+         * The idea here is to consume whole bytes at a time rather than individual bits. node
+         * represents the Huffman tree, with all bit patterns denormalized as 256 children. Each
+         * child represents the last 8 bits of the huffman code. The parents of each child each
+         * represent the successive 8 bit chunks that lead up to the last most part. 8 bit bytes
+         * from buf are used to traverse these tree until a terminal node is found.
+         *
+         * current is a bit buffer. The low order bits represent how much of the huffman code has
+         * not been used to traverse the tree. Thus, the high order bits are just garbage.
+         * currentBits represents how many of the low order bits of current are actually valid.
+         * currentBits will vary between 0 and 15.
+         *
+         * symbolBits is the number of bits of the the symbol being decoded, *including* all those
+         * of the parent nodes. symbolBits tells how far down the tree we are. For example, when
+         * decoding the invalid sequence {0xff, 0xff}, currentBits will be 0, but symbolBits will be
+         * 16. This is used to know if buf ended early (before consuming a whole symbol) or if
+         * there is too much padding.
+         */
+        @Override
+        public boolean process(byte value) throws IOException {
+            int b = value & 0xFF;
+            current = (current << 8) | b;
+            currentBits += 8;
+            symbolBits += 8;
+            // While there are unconsumed bits in current, keep consuming symbols.
+            while (currentBits >= 8) {
+                int c = (current >>> (currentBits - 8)) & 0xFF;
+                node = node.children[c];
+                currentBits -= node.bits;
+                if (node.isTerminal()) {
+                    if (node.symbol == HpackUtil.HUFFMAN_EOS) {
+                        throw EOS_DECODED;
+                    }
+                    append(node.symbol);
+                    node = ROOT;
+                    // Upon consuming a whole symbol, reset the symbol bits to the number of bits
+                    // left over in the byte.
+                    symbolBits = currentBits;
+                }
+            }
+            return true;
+        }
+
+        AsciiString end() throws IOException {
+            /*
+             * We have consumed all the bytes in buf, but haven't consumed all the symbols. We may be on
+             * a partial symbol, so consume until there is nothing left. This will loop at most 2 times.
+             */
+            while (currentBits > 0) {
+                int c = (current << (8 - currentBits)) & 0xFF;
```

Comment:
consider removing `c` and just inline the arithmetic below ... `c` is only used a single time.